### PR TITLE
Add travis configuration

### PR DESCRIPTION
**Pull request by TheBB**
@VikingScientist figured you'd like this. :-P

This runs tests on Travis for three operating systems and four Python versions. Massive thanks to Miniconda without which this would have been really tricky.

Note: this PR includes #109 which enables the cython module.

I was unable to get OpenCV to work properly on Windows so those tests don't run there.

Sample build: https://travis-ci.org/TheBB/Splipy/builds/652992226

I'll use [cibuildwheel](https://github.com/joerick/cibuildwheel) to build wheels for pypi as well. Then this should be ready.

### CODE REVIEW

**Comment by TheBB** [comment]
This is ready now. I created a package called 'Splipy-TestPackage' to test the deployment. See [here](https://pypi.org/project/Splipy-TestPackage/#files).

To enable upload to PyPi you have to go into Travis settings and add a TWINE_PASSWORD environment variable. Set it equal to an API token for PyPi that has access to upload packages to Splipy. Since I have access to both these places I've already done this.

To make a new release, all you should need to do is to tag a version on github. However it's the version number in the setup.py file that matters for PyPi!

Let me know if you have questions.

**Comment by TheBB** [comment]
Figured out why github is not notified about the build. Apparently Travis doesn't have access to sintefmath.

Screenshot from 2020-02-21 12-34-58

I made a request for it so maybe someone is listening on the other end?

Note this doesn't seem to affect the actual builds. They still run. It's just that the results aren't being displayed here on github.

**Comment by VikingScientist** [comment]
Access granted

File: README.rst [hunk 1]
```diff
@@ -120,17 +122,6 @@ number, create a commit and a tag. To push this to github, use::
...
-they need to be uploaded. The recommended way to do that is using `twine
-<https://pypi.python.org/pypi/twine>`_::
-
-    twine upload dist/* -r <index>
-
-Where `index` is the name of the index in your `~/.pypirc` where you want to
-upload.
+After that, Travis CI should automatically build and deploy the
+packages to PyPi. It would be helpful to monitor the Travis build so
+that errors can be fixed quickly.
```
Comment (VikingScientist) [inline README.rst]: Where and how do you monitor this build? Do you have a direct link?

Comment (TheBB) [reply README.rst]: https://travis-ci.org/sintefmath/Splipy/builds

**Comment by TheBB** [comment]
Ah beautiful. Thanks. Now you can see the test status on github.

File: README.rst (hunk 1 above)
Comment (TheBB) [reply README.rst]: Added link in readme.

**Pull request merged**